lib/mongodb.js

```diff
@@ -112,6 +112,25 @@ exports.initialize = function initializeDataSource(dataSource, callback) {
   }
 };
 
+exports.fieldsArrayToObj = fieldsArrayToObj;
+
+/**
+ * Helper function to accept an array representation of fields projection and return the mongo
+ * required object notation
+ *
+ * @param {string[]} fieldsArray
+ * @returns {Object}
+ */
+function fieldsArrayToObj(fieldsArray) {
+  if (!Array.isArray(fieldsArray)) return fieldsArray; // fail safe check in case non array object created
+  return fieldsArray.length ?
+    fieldsArray.reduce(function arrayToObjectReducer(result, field) {
```

Comment:
Can we move this function to the outer scope (outside of `fieldsArrayToObj`), so that V8 does not have to create a new function every time `fieldsArrayToObj` is called?

```js
function fieldsArrayToObj(fieldsArray) {
  // ...
}

function arrayToObjectReducer(result, field) {
  // ...
}
```

Reply:
You are right. Done!